homestar-runtime/src/settings.rs

```diff
@@ -22,6 +24,55 @@ pub struct Monitoring {
     process_collector_interval: u64,
 }
 
+#[derive(Default, Clone, Debug, Deserialize)]
+/// Configure how the Network keypair is generated or using an existing one
+pub(crate) enum PubkeyConfig {
+    #[default]
+    #[serde(rename = "random")]
+    Random,
+    /// Seed string is hashed with SHA-256 to produce the ed25519 secret key.
+    #[serde(rename = "seed")]
+    GenerateFromSeed(String),
+    /// File path to a PEM encoded ed25519 key
+    #[serde(rename = "path")]
+    Existing(String),
+}
+
+
+impl PubkeyConfig {
+    /// Produce a Keypair using the given configuration. Consumes `self` to avoid keeping secrets laying around.
+    fn generate_keypair(self) -> Result<identity::Keypair, DecodingError> {
+        match self {
+            PubkeyConfig::Random => Ok(identity::Keypair::generate_ed25519()),
+            PubkeyConfig::GenerateFromSeed(seed) => {
+                let mut hasher = Sha256::default();
+                hasher.update(&seed);
+                identity::Keypair::ed25519_from_bytes(hasher.finalize())
+            },
+            PubkeyConfig::Existing(path) => {
+                let path = Path::new(&path);
+                println!("{:?}", path);
```

Comment:
guessing prints are just for testing. We do have `tracing` enabling for logging.

Reply:
yep, will add logs for tracing about the same time as the actual tests